core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/JwtAuthRule.scala

```diff
@@ -71,10 +72,10 @@ final class JwtAuthRule(val settings: JwtAuthRule.Settings,
           case RequestGroup.`N/A` =>
             authorizeUsingJwtToken(blockContext)
           case RequestGroup.AGroup(group) =>
-            settings.permittedGroups.toList match {
-              case Nil =>
+            settings.groupsLogic match {
+              case GroupsLogic.NotDefined =>
                 authorizeUsingJwtToken(blockContext)
-              case permittedGroups if permittedGroups.contains(group) =>
+              case GroupsLogic.Defined(strategy) if strategy.availableGroupsFrom(UniqueList.of(group)).isDefined =>
```

Comment:
`strategy.groups.contains(group)` we don't need to check here based on strategy (so the `groups` extension method will be needed)